libs/util/shared/src/lib/links/link-utils.ts

```diff
@@ -242,6 +251,47 @@ export function getLinkLabel(link: DatasetOnlineResource): string {
   return format ? `${label} (${format})` : label
 }
 
+export async function getLayers(url: string, serviceProtocol: ServiceProtocol) {
+  switch (serviceProtocol) {
+    case 'ogcFeatures': {
+      const layers = await new OgcApiEndpoint(url).allCollections
+      return layers
+    }
+    case 'wfs': {
+      const endpointWfs = new WfsEndpoint(url)
+      await endpointWfs.isReady()
+      return endpointWfs.getFeatureTypes()
+    }
+    case 'wms': {
+      const endpointWms = new WmsEndpoint(url)
+      await endpointWms.isReady()
+      return endpointWms
+        .getLayers()
+        .flatMap(wmsLayerFlatten)
+        .filter((l) => l.name)
+    }
+    case 'wmts': {
+      const endpointWmts = new WmtsEndpoint(url)
+      await endpointWmts.isReady()
```

Comment:
FYI you can also do `const endpoint = await new WmtsEndpoints(url).isReady()`

No need to change it here, just for next time ;)